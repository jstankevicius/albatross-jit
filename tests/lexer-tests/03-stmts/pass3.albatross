if (0) { while(0) { return 0; } otherwise { return 0 := 0; }
while (0) { if (0) { 0+0; } else { 0-0; } } otherwise { return 0; }