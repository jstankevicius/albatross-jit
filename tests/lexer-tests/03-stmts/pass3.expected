1 1 IF
4 1 (
5 1 INT 0
6 1 )
8 1 {
10 1 WHILE
15 1 (
16 1 INT 0
17 1 )
19 1 {
21 1 RETURN
28 1 INT 0
29 1 SEMICOLON
31 1 }
33 1 OTHERWISE
43 1 {
45 1 RETURN
52 1 INT 0
54 1 ASSIGN
57 1 INT 0
58 1 SEMICOLON
60 1 }
1 2 WHILE
7 2 (
8 2 INT 0
9 2 )
11 2 {
13 2 IF
16 2 (
17 2 INT 0
18 2 )
20 2 {
22 2 INT 0
23 2 PLUS
24 2 INT 0
25 2 SEMICOLON
27 2 }
29 2 ELSE
34 2 {
36 2 INT 0
37 2 MINUS
38 2 INT 0
39 2 SEMICOLON
41 2 }
43 2 }
45 2 OTHERWISE
55 2 {
57 2 RETURN
64 2 INT 0
65 2 SEMICOLON
67 2 }
